test/test_autograd.py

```diff
@@ -12647,6 +12647,9 @@ def functional_op(x):
         self.assertFalse(func_out.requires_grad)
         self.assertTrue(func_out.is_leaf)
 
+    @skipIfTorchDynamo(
+        "exception from ill-formed graph module is not propagated with eager_noexcept"
```

Comment:
do you know why we needed this skip? (does this point to us giving a worse error message on inference errors than we used to?)

Reply:
Its just .. an annoying reason. Dynamo wrapped tests work with `eager_noexcept` backend. The idea is that if Dynamo generates a bad graph, we want to raise a TorchDynamo error right then and not let the bad graph go further and fails for some silly reason later on. 

This test relies on the function (which converts to Fx graph for Dynamo) to raise an exception. However, Dynamo thinks that it has formed a bad Fx graph and therefore returns a DynamoException instead of RuntimeException.

Reply:
![image](https://github.com/user-attachments/assets/5dc41ac6-1925-4c07-8600-0c0dcc4cd676)

Reply:
ah ok. So I guess we get the same error in eager and compile, just it shows up to the user as a `DynamoException`?

Reply:
Correct.